security-framework/src/certificate.rs

```diff
@@ -57,8 +83,124 @@ impl SecCertificate {
             CFString::wrap_under_create_rule(summary).to_string()
         }
     }
+
+    #[cfg(any(feature = "OSX_10_12", target_os = "ios"))]
+    /// Returns DER encoded subjectPublicKeyInfo of certificate if available. This can be used
+    /// for certificate pinning.
+    pub fn public_key_info_der(&self) -> Result<Option<Vec<u8>>> {
+        // Imported from TrustKit
+        // https://github.com/datatheorem/TrustKit/blob/master/TrustKit/Pinning/TSKSPKIHashCache.m
+        let public_key = self.public_key()?;
+        let public_key_attributes = public_key.attributes();
+        let public_key_type =
+            public_key_attributes.find(unsafe { kSecAttrKeyType } as *const std::os::raw::c_void);
```

Comment:
`kSecAttrKeyType` is a constant right? So `unsafe { kSecAttrKeyType }` shouldn't need unsafe.

Reply:
It's apparently a new thing by the compiler:

error: use of extern static is unsafe and requires unsafe function or block (error E0133)
 note: #[deny(safe_extern_statics)] on by default
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

Reply:
Ah, right. It's not a (Rust) constant, but technically a global mutable variable. So unsafe makes sense!